src/foam/nanos/auth/PasswordExpiryAuthService.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.auth',
+  name: 'PasswordExpiryAuthService',
+  extends: 'foam.nanos.auth.ProxyAuthService',
+
+  documentation: 'Checks password expiry',
+
+  imports: [
+    'localUserDAO'
+  ],
+
+  javaImports: [
+    'javax.naming.AuthenticationException',
+    'java.util.Calendar',
+    'java.util.TimeZone'
+  ],
+
+  methods: [
+    {
+      name: 'login',
+      javaCode:
+`User user = getDelegate().login(x, userId, password);
+ if ( isPasswordExpired(user) ) {
+   throw new AuthenticationException("Password expired");
+ }
+ return user;`
+    },
+    {
+      name: 'loginByEmail',
+      javaCode:
+`User user = getDelegate().loginByEmail(x, email, password);
```

Comment:
This will end up creating a session and putting into the session dao even though we're rejecting the login due to stale password. Are we sure this is the behaviour we want?

Reply:
Good point.